src/util/timestamper.cpp

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************\
+
+Module: Timestamps
+
+Author: Kareem Khazem <karkhaz@karkhaz.com>
+
+\*******************************************************************/
+
+
+#include <chrono>
+#include <cstdlib>
+
+#include "invariant.h"
+#include "timestamper.h"
+
+std::unique_ptr<const timestampert>
+timestampert::make(timestampert::clockt clock_type)
+{
+#ifdef _WIN32
+  return std::unique_ptr<const timestampert>(new timestampert());
+#else
+  switch(clock_type)
+  {
+    case timestampert::clockt::NONE:
+      return std::unique_ptr<const timestampert>(new timestampert());
+    case timestampert::clockt::MONOTONIC:
+      return std::unique_ptr<const monotonic_timestampert>(
+          new monotonic_timestampert());
+    case timestampert::clockt::WALL_CLOCK:
+      return std::unique_ptr<const wall_clock_timestampert>(
+          new wall_clock_timestampert());
+  }
+  UNREACHABLE;
+#endif
+}
+
+#ifndef _WIN32
+std::string monotonic_timestampert::stamp() const
+{
+  std::chrono::time_point<std::chrono::steady_clock, std::chrono::microseconds>
+    time_stamp = std::chrono::time_point_cast<std::chrono::microseconds>(
+      std::chrono::steady_clock::now());
+
+  auto cnt = time_stamp.time_since_epoch().count();
+  std::div_t divmod = div(cnt, 1000000);
+  char u_seconds_fmt[7];
+  std::snprintf(u_seconds_fmt, sizeof(u_seconds_fmt), "%06d", divmod.rem);
```

Comment:
How about `std::setfill('0') << std::setw(6) << divmod.rem` to get rid of `snprintf`?